R/PackageFunctionReporter.R

```diff
@@ -185,40 +183,98 @@ FunctionReporter <- R6::R6Class(
 
             log_info(sprintf('Constructing network representation...'))
 
-            # foodweb will output a warning for "In par(oldpar) : calling par(new=TRUE) with no plot" all the time.
-            # does not seem to be an issue
-            funcMap <- suppressWarnings({
-                mvbutils::foodweb(
-                    where = paste("package"
-                                  , self$pkg_name
-                                  , sep = ":")
-                    ,
-                    plotting = FALSE
-                )
-            })
+            # Get table of edges between functions
+            edgeDT <- .get_edges(pkg_name = self$pkg_name)
 
             log_info("Done constructing network representation")
 
-            # Function Connections: Edges
-            edges <- data.table::melt(
-                data.table::as.data.table(funcMap$funmat, keep.rownames = TRUE)
-                , id.vars = "rn"
-            )[value != 0]
-
-            # Formatting
-            edges[, value := NULL]
-            edges[, SOURCE := as.character(variable)]
-            edges[, TARGET := as.character(rn)]
-            edges[, variable := NULL]
-            edges[, rn := NULL]
-            data.table::setcolorder(edges, c('SOURCE', 'TARGET'))
-
-            # If no edges, return empty data.table
-            if (nrow(edges) == 0) {
-                return(data.table::data.table(SOURCE = character(), TARGET = character()))
-            }
-
-            return(edges)
+            return(edgeDT)
         }
     )
 )
+
+#' @importFrom data.table rbindlist
+.get_edges <- function(pkg_name){
```

Comment:
I don't like the idea of calling this function `.get_edges` and putting it in the main package environment (outside of `FunctionReporter`). You should name it something more descriptive, like `.get_function_graph_edges`.

Reply:
oh fair, can do that!